src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -0,0 +1,70 @@
+package frc.robot.subsystems;
+
+import com.spikes2212.command.DashboardedSubsystem;
+import com.spikes2212.util.Limelight;
+import org.photonvision.PhotonCamera;
+import org.photonvision.targeting.PhotonPipelineResult;
+
+public class Vision extends DashboardedSubsystem {
+
+    public static final String PHOTON_VISION_CAMERA_NAME = "photoncamera";
+
+    public static final int PIPELINE_CONE_INDEX = 0;
+    public static final int PIPELINE_CUBE_INDEX = 1;
+
+    public static final int PIPELINE_RETROREFLECTIVE_INDEX = 0;
+    public static final int PIPELINE_APRILTAG_INDEX = 1;
+
+    private static Vision instance;
+
+    private final Limelight limelight;
+    private final PhotonCamera photonCamera;
+
+    public static Vision getInstance(){
+        if (instance==null){
+            instance = new Vision(new PhotonCamera(PHOTON_VISION_CAMERA_NAME), new Limelight());
+        }
+        return instance;
+    }
+
+    private Vision(PhotonCamera photonCamera, Limelight limelight) {
+        super("vision");
```

Comment:
get the namespaceName as a parameter and give it when you construct the instance